### ISSUE TITLE
Illegal istruction when running dart2js in a docker container 

### ISSUE THREAD

**Issue by peppetort** [flutter/flutter#94181]
We had a problem while using an old pc as a build server using a docker image. In particular we get an illegal istruction when running `flutter build web`
The CPU on the build server is a Intel(R) Atom(TM) CPU  330   @ 1.60GHz

Are there any minimal CPU requirements to run flutter build?
Thanks

## Steps to Reproduce

1. Create a new project
2. Run flutter build web

## Logs

<details>
<summary>Logs</summary>

```
💪 Building with sound null safety 💪

Compiling lib/main.dart for the Web...                          
Target dart2js failed: Exception: 
===== CRASH =====
si_signo=Illegal instruction(4), si_code=2, si_addr=0x7f21de736887
version=2.16.0-38.0.dev (dev) (Mon Nov 22 14:23:59 2021 -0800) on "linux_x64"
pid=40, thread=46, isolate_group=main(0x55b373489000), isolate=main(0x55b37348b000)
isolate_instructions=7f21de5b5000, vm_instructions=55b370ccd640
  pc 0x00007f21de736887 fp 0x00007f21d88fd0a8 Unknown symbol
  pc 0x00007f21de739e16 fp 0x00007f21d88fd0e0 Unknown symbol
  pc 0x00007f21de739a29 fp 0x00007f21d88fd120 Unknown symbol
  pc 0x00007f21df53fa1a fp 0x00007f21d88fd208 Unknown symbol
  pc 0x00007f21df5436a5 fp 0x00007f21d88fd288 Unknown symbol
  pc 0x00007f21df4a7b54 fp 0x00007f21d88fd2b8 Unknown symbol
  pc 0x00007f21df5447e3 fp 0x00007f21d88fd310 Unknown symbol
  pc 0x00007f21de952a95 fp 0x00007f21d88fd368 Unknown symbol
  pc 0x00007f21df566da7 fp 0x00007f21d88fd3f0 Unknown symbol
  pc 0x00007f21df56cfdb fp 0x00007f21d88fd420 Unknown symbol
  pc 0x00007f21df5447e3 fp 0x00007f21d88fd478 Unknown symbol
  pc 0x00007f21de759231 fp 0x00007f21d88fd4e8 Unknown symbol
  pc 0x00007f21de75ad38 fp 0x00007f21d88fd538 Unknown symbol
  pc 0x00007f21de75b51a fp 0x00007f21d88fd588 Unknown symbol
  pc 0x00007f21df665a2c fp 0x00007f21d88fd5f8 Unknown symbol
  pc 0x00007f21de75b218 fp 0x00007f21d88fd658 Unknown symbol
  pc 0x00007f21df665849 fp 0x00007f21d88fd6b8 Unknown symbol
  pc 0x00007f21df6c799a fp 0x00007f21d88fd710 Unknown symbol
  pc 0x00007f21de766d55 fp 0x00007f21d88fd778 Unknown symbol
  pc 0x00007f21df3991a1 fp 0x00007f21d88fd7c0 Unknown symbol
  pc 0x00007f21df66981c fp 0x00007f21d88fd808 Unknown symbol
  pc 0x00007f21de75b218 fp 0x00007f21d88fd868 Unknown symbol
  pc 0x00007f21df66934e fp 0x00007f21d88fd8b8 Unknown symbol
  pc 0x00007f21df6c85da fp 0x00007f21d88fd910 Unknown symbol
  pc 0x00007f21de766d55 fp 0x00007f21d88fd978 Unknown symbol
  pc 0x00007f21df39979b fp 0x00007f21d88fd9d0 Unknown symbol
  pc 0x00007f21df399ef8 fp 0x00007f21d88fda20 Unknown symbol
  pc 0x00007f21df39ad69 fp 0x00007f21d88fdad8 Unknown symbol
  pc 0x00007f21df39b931 fp 0x00007f21d88fdb88 Unknown symbol
  pc 0x00007f21df3b1743 fp 0x00007f21d88fdc18 Unknown symbol
  pc 0x00007f21de715b79 fp 0x00007f21d88fdc90 Unknown symbol
  pc 0x00007f21de715e8c fp 0x00007f21d88fdcf8 Unknown symbol
  pc 0x00007f21df58a226 fp 0x00007f21d88fdd90 Unknown symbol
  pc 0x00007f21df58a42e fp 0x00007f21d88fdde0 Unknown symbol
  pc 0x00007f21df58beae fp 0x00007f21d88fde40 Unknown symbol
  pc 0x00007f21bb0baeb7 fp 0x00007f21d88fdea8 Unknown symbol
  pc 0x00007f21df47f9eb fp 0x00007f21d88fdef0 Unknown symbol
  pc 0x00007f21df709f05 fp 0x00007f21d88fdf30 Unknown symbol
  pc 0x00007f21df3b10d9 fp 0x00007f21d88fdfd0 Unknown symbol
  pc 0x00007f21de715b79 fp 0x00007f21d88fe048 Unknown symbol
  pc 0x00007f21de715e8c fp 0x00007f21d88fe0b0 Unknown symbol
  pc 0x00007f21df58a226 fp 0x00007f21d88fe148 Unknown symbol
  pc 0x00007f21df58a42e fp 0x00007f21d88fe198 Unknown symbol
  pc 0x00007f21df58beae fp 0x00007f21d88fe1f8 Unknown symbol
  pc 0x00007f21bb0baeb7 fp 0x00007f21d88fe260 Unknown symbol
  pc 0x00007f21df47f9eb fp 0x00007f21d88fe2a8 Unknown symbol
  pc 0x00007f21df709f05 fp 0x00007f21d88fe2e8 Unknown symbol
  pc 0x00007f21df3b08c0 fp 0x00007f21d88fe388 Unknown symbol
  pc 0x00007f21de715b79 fp 0x00007f21d88fe400 Unknown symbol
  pc 0x00007f21de715e8c fp 0x00007f21d88fe468 Unknown symbol
  pc 0x00007f21df58a226 fp 0x00007f21d88fe500 Unknown symbol
  pc 0x00007f21df58a42e fp 0x00007f21d88fe550 Unknown symbol
  pc 0x00007f21df58beae fp 0x00007f21d88fe5b0 Unknown symbol
  pc 0x00007f21bb0baeb7 fp 0x00007f21d88fe618 Unknown symbol
  pc 0x00007f21df47f9eb fp 0x00007f21d88fe660 Unknown symbol
  pc 0x00007f21df58d42d fp 0x00007f21d88fe6a0 Unknown symbol
  pc 0x00007f21df58ff51 fp 0x00007f21d88fe710 Unknown symbol
  pc 0x00007f21df590129 fp 0x00007f21d88fe770 Unknown symbol
  pc 0x00007f21de716853 fp 0x00007f21d88fe800 Unknown symbol
  pc 0x00007f21df5903e0 fp 0x00007f21d88fe850 Unknown symbol
  pc 0x00007f21df595ac3 fp 0x00007f21d88fe898 Unknown symbol
  pc 0x00007f21df59653b fp 0x00007f21d88fe8d8 Unknown symbol
  pc 0x00007f21df596759 fp 0x00007f21d88fe900 Unknown symbol
  pc 0x00007f21df709d6e fp 0x00007f21d88fe940 Unknown symbol
  pc 0x00007f21df70a8d8 fp 0x00007f21d88fe980 Unknown symbol
  pc 0x00007f21de0029af fp 0x00007f21d88fe9f8 Unknown symbol
  pc 0x000055b370e43153 fp 0x00007f21d88feaa0 dart::DartEntry::InvokeCode(dart::Code const&, unsigned long, dart::Array const&, dart::Array const&, dart::Thread*)+0x153
  pc 0x000055b370e42f8c fp 0x00007f21d88feb00 dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array const&, dart::Array const&, unsigned long)+0x14c
  pc 0x000055b370e454ff fp 0x00007f21d88feb50 dart::DartLibraryCalls::HandleMessage(long, dart::Instance const&)+0x16f
  pc 0x000055b370e6c394 fp 0x00007f21d88fec30 dart::IsolateMessageHandler::HandleMessage(std::__2::unique_ptr<dart::Message, std::__2::default_delete<dart::Message> >)+0x374
  pc 0x000055b370e9426c fp 0x00007f21d88feca0 dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)+0x14c
  pc 0x000055b370e9496f fp 0x00007f21d88fecf0 dart::MessageHandler::TaskCallback()+0x1df
  pc 0x000055b370fac848 fp 0x00007f21d88fed80 dart::ThreadPool::WorkerLoop(dart::ThreadPool::Worker*)+0x138
  pc 0x000055b370faccdc fp 0x00007f21d88fedb0 dart::ThreadPool::Worker::Main(unsigned long)+0x5c
  pc 0x000055b370f28018 fp 0x00007f21d88fee70 /usr/local/flutter/bin/cache/dart-sdk/bin/dart+0x2028018
-- End of DumpStackTrace

Compiling lib/main.dart for the Web...                             87.2s
Exception: Failed to compile application for the Web.
```

```
[✓] Flutter (Channel master, 2.6.0-12.0.pre.819, on Debian GNU/Linux 11 (bullseye) 5.4.0-90-generic, locale en_US)
    • Flutter version 2.6.0-12.0.pre.819 at /usr/local/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a1115b8b3f (20 hours ago), 2021-11-23 14:49:13 -0800
    • Engine revision 41b8e57bc4
    • Dart version 2.16.0 (build 2.16.0-38.0.dev)
    • DevTools version 2.8.0

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✗] Chrome - develop for the web (Cannot find Chrome executable at google-chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[!] Connected device
    ! No devices available

! Doctor found issues in 4 categories.

```

</details>

**Comment by mraleph** [comment]
Most likely one of the SSE4 instructions. Might be `roundsd`. It would help if you take a core dump on the build server and then look at which instruction crashed.

One potential reason is that we generate app-jit snapshots on a newer CPUs but potentially use them on older CPUs - so we should either tell VM not to use optional CPU features (e.g. SSE4.1) or remember which features it used for a specific code so that we could invalidate that code when loading it into the snapshot.

/cc @mkustermann

**Comment by alexmarkov** [comment]
Possible fix was rolled into Flutter in 45808bab256d92a5cf676eee8055e0e79e4e2796.
@peppetort Could you verify if the crash disappears on the latest Flutter?

**Comment by peppetort** [comment]
I just checked and with the latest version there are no more problems.
Thank you
